### ISSUE TITLE
[Bug] What does .submit do? What can it return/what errors can it throw? Astounding lack of documentation

### ISSUE THREAD

**Issue by KayakinKoder** [paypal/paypal-js#554]
### Library used

All:

### 🐞 Describe the Bug

The js SDKs all have calling `.submit` as a first step. Yet there is _no_ documentation on this function, which is just astounding. How do you expect anyone to build a halfway decent integration with your SDK without _basic_ documentation?

### 🔬 Minimal Reproduction

NA

### 😕 Actual Behavior

NA

### 🤔 Expected Behavior

Basic documentation.

### 🌍 Environment

NA

### ➕ Additional Context

We are regretting our choice of PayPal. We're sitting here _guessing_ what this code does, what it can return, when and why it can throw, etc. trying to build a checkout page.

**Comment by KayakinKoder** [comment]
@hnguyen510 @spencersablan @wsbrunson sorry for randomly tagging you all, but we (paying customers for 10+ years) need _someone_ to provide technical support. 

Nobody responds to issues opened in this repo. So, where do we go/who do we ask for technical support?

Very basic things like the `.submit` function are not documented anywhere we can find. We're sitting here wasting days upon days manually testing with real credit cards to try and figure out how this SDK works.

Where do we go/who do we ask for actual help here? This is extremely, extremely frustrating.

**Comment by KayakinKoder** [comment]
A warning to anyone considering PayPal for payments integrations -- just don't do it. We opened a technical support ticket 11 days ago, asked questions here in this repo, and PayPal's "support" is nonexistent. They're unwilling or incapable of answering basic questions about their APIs/SDKs.

Comparing Stripe's documentation, support, Developer Workbench, etc. is a night and day difference. We're switching as quickly as we possibly can.

**Comment by wsbrunson** [comment]
@KayakinKoder what feature of the SDK are you trying to integrate with? Which doc page are you using?

**Comment by KayakinKoder** [comment]
- Online Checkout Advanced (https://developer.paypal.com/studio/checkout/advanced), React SDK (https://github.com/paypal/paypal-js/tree/main/packages/react-paypal-js)
- The "documentation" is here: https://developer.paypal.com/sdk/js/reference/#link-submitpromisevoid
![image](https://github.com/user-attachments/assets/1beddb08-fe05-4ad3-86d7-4a32311e38ce)

I fail to see how anyone could build an even remotely decent integration without knowing:
- All of the examples show logging the error to the console. Is that _really_ what you recommend we do in production?
- If you recommend we actually display an error to the user in production, it might be helpful to know things like: what does actually submit do? What errors can it throw/in what situations will it throw? Are these recoverable errors (e.g. the user input their CVV incorrectly) that the user can simply edit the form and click the submit button again, or are unrecoverable errors? In a nutshell, what should we as developers using your SDK _actually do_ when submit does not work as expected?

**Comment by wsbrunson** [comment]
PayPal's Card Fields use the `.submit` function to trigger the create order -> buyer approval -> capture payment flow. The merchant is responsible for rendering their own submit button and also implementing keyboard shortcuts like pressing enter to submit. This is so the merchant has full control over their forms on their pages, PayPal is just handling the PCI-sensitive fields or fields we need for risk assessment.

PayPal's SDK relies on callbacks passed to the component, the main ones being:

* `createOrder`
* `onApprove`
* `onCancel`
* `onError`

The SDK will trigger these callbacks at certain points during the lifecycle of the payment. Merchant can then take action depending on which callback was invoked. In a happy-path scenario, `createOrder` will be invoked on click, and `onApprove` will be invoked to let the merchant know to capture payment.

`.submit` is promisified so you are able to await it's result and also catch any errors from it. We suggest using the callbacks above though.

> All of the examples show logging the error to the console. Is that really what you recommend we do in production?

We don't but that is strictly up to the merchant in how they would prefer to alert buyers to any issues. We are only trying to provide PCI-compliant card fields; merchants should decide how they would like to display errors.

> If you recommend we display an error to the user in production...

These are good suggestions and something we've been thinking about implementing. I'll take this feedback to our product team.

> ...does a new order need to be created or can the order that was created when submit was clicked be used...

I would recommend creating a new order when using a new payment method. I can ask our Orders API team what would happen if a merchant tried to apply a different payment method to an order at various points during the order lifecycle in the meantime.

**Comment by KayakinKoder** [comment]
Thank you.

> .submit is promisified so you are able to await it's result and also catch any errors from it. We suggest using the callbacks above though.

Do you mean that you recommend ignoring errors from `.submit`  (therefore we don't really need to know _what_ errors it can throw) because the createOrder/onApprove workflow should still occur even when `.submit` throws?

**Comment by wsbrunson** [comment]
Yes exactly. The `.submit` being promisified was to allow backwards-compatibility with our earlier v1 Hosted Fields product that was replaced by the Card Fields product. The old product was kind of a hybrid between the way Braintree did unbranded card fields and the way the PayPal SDK offered components.

The new Card Fields were written to look much more like a PayPal SDK component but there are some parts that stuck around to make it easier for merchants to migrate.

I suggested that we even remove the `.submit.then.error` example, I am going to revisit that with our product and docs team
